lib/types.ts

```diff
@@ -131,7 +139,44 @@ export interface EntryFieldAPI {
   getForLocale: (locale: string) => FieldAPI
 }
 
-export interface EntryAPI {
+type TaskState = 'active' | 'resolved'
+
+export interface TaskSys {
+  id: string
+  type: 'Task'
+  parentEntity: { sys: Link }
+  space: Link
+  environment: Link
+  createdBy: Link
+  createdAt: string
+  updatedBy: Link
+  updatedAt: string
+  version: number
+}
+
+export interface TaskInputData {
+  assignedToId: string
+  body: string
+  status: TaskState
+}
+
+export interface Task {
+  assignedTo: Link
+  body: string
+  status: TaskState
+  sys: TaskSys
+}
+
+/** Allows accessing the Task API for the current entry. */
+export interface TaskAPI {
+  getTask(id: string): Promise<Task>
+  getTasks(): Promise<CollectionResponse<Task>>
+  createTask(data: TaskInputData): Promise<Task>
+  updateTask(task: Task): Promise<Task>
+  deleteTask(task: Task): Promise<void>
+}
+
+export interface EntryAPI extends TaskAPI {
```

Comment:
Not happy with extending `EntryAPI` here as we can only extend from one interface. This will become problematic when adding new apis to the entry in the future.

I'd rather add all functions of `TaskAPI` directly into `EntryAPI`

Reply:
`interface A extends B, C, D {}` works?
And the reason for the split is that it made it easier to add the methods in `entry.ts`
```js
  const taskApi = {} as TaskAPI
  ...
  return {
    otherThings,
   ...taskApi
   }
```
But it's also easily possible to do this without splitting the interface, so if you still feel strongly about changing it let me know

Reply:
Oh - didn't know extending multiple interfaces